src/components/transformation/create/Config/SQLEditor/useSQLEditor.ts

```diff
@@ -0,0 +1,142 @@
+    const draftId = useEditorStore_persistedDraftId();
+    const draftSpecs = useEditorStore_queryResponse_draftSpecs();
+    const mutateDraftSpecs = useEditorStore_queryResponse_mutate();
+
+    const currentCatalog = useEditorStore_currentCatalog();
+    const setSpecs = useEditorStore_setSpecs();
+
+    // Transform Create Store
+    const transformConfigs = useTransformationCreate_transformConfigs();
+    const migrations = useTransformationCreate_migrations();
+    const attributeType = useTransformationCreate_attributeType();
+    const attributeId = useTransformationCreate_selectedAttribute();
+    const patchSelectedAttribute =
+        useTransformationCreate_patchSelectedAttribute();
+
+    const [draftSpec, setDraftSpec] = useState<DraftSpec>(null);
+
+    const defaultSQL = useMemo(() => {
+        if (
+            attributeType === 'transform' &&
+            Object.hasOwn(transformConfigs, attributeId)
+        ) {
+            return transformConfigs[attributeId].lambda;
+        } else if (
+            attributeType === 'migration' &&
+            Object.hasOwn(migrations, attributeId)
+        ) {
+            return migrations[attributeId];
+        } else {
+            return '';
+        }
+    }, [attributeType, migrations, transformConfigs, attributeId]);
+
+    const processEditorValue = useCallback(
+        async (
+            value: any,
+            catalogName: string,
+            specType: Entity,
+            propUpdating?: AllowedScopes | string
+        ) => {
+            if (!mutateDraftSpecs || !draftSpec) {
+                return Promise.reject();
+            } else {
+                if (propUpdating) {
+                    const evaluatedAttribute =
+                        attributeType === 'transform'
+                            ? updateTransforms(
+                                  transformConfigs[attributeId].name,
+                                  value,
+                                  transformConfigs
+                              )
+                            : updateMigrations(attributeId, value, migrations);
+
+                    set(draftSpec.spec, propUpdating, evaluatedAttribute);
+                } else {
+                    draftSpec.spec = value;
+                }
+
+                const updateResponse = await modifyDraftSpec(draftSpec.spec, {
+                    draft_id: draftId,
+                    catalog_name: catalogName,
+                    spec_type: specType,
+                });
+
+                if (updateResponse.error) {
+                    return Promise.reject();
+                }
+
+                patchSelectedAttribute(value);
+
+                return mutateDraftSpecs();
+            }
+        },
+        [
+            patchSelectedAttribute,
+            attributeId,
+            attributeType,
+            draftId,
+            draftSpec,
+            migrations,
+            mutateDraftSpecs,
+            transformConfigs,
+        ]
+    );
+
```

Comment:
I think these effects are the same as in the `useDraftSpecEditor` if these are going to be combines in the future please add a comment stating that. Otherwise, please make 'em share this functionality.